plugin/src/lib.rs

```diff
@@ -19,14 +20,44 @@ use syntax::ptr::P;
 use syntax::symbol::Symbol;
 use syntax::util::small_vector::SmallVector;
 use syntax::ast::{IntTy, LitIntType, LitKind, UnOp};
+use syntax::ext::base::MultiItemModifier;
 
 mod binop;
+mod bounded_loop;
+
+
+/// ChainedMultiMutator is a MultiMutator which allows to chain two `MultiItemModifier`.
+struct ChainedMultiMutator {
+    left: Box<MultiItemModifier>,
+    right: Box<MultiItemModifier>,
+}
+
+impl MultiItemModifier for ChainedMultiMutator {
+    fn expand(&self, cx: &mut ExtCtxt, span: Span, mi: &MetaItem, a: Annotatable) -> Vec<Annotatable> {
+        let out = self.left.expand(cx, span, mi, a);
+
+        out.into_iter()
+            .map(|a| self.right.expand(cx, span, mi, a))
+            .collect::<Vec<Vec<Annotatable>>>()
+            .into_iter()
+            .fold(Vec::new(), |mut acc, outs| {
+                acc.extend(outs);
+
+                acc
+            })
+    }
+}
 
 #[plugin_registrar]
 pub fn plugin_registrar(reg: &mut Registry) {
```

Comment:
I'm not sure if I like to split the functionality this way, but I don't have a too strong opinion about it, so this is not something that would keep me from merging.

Reply:
I like the fact of having the 2 plugins separated. Even that, I would like to hear for alternatives. Or we can change this on the (near) future.